src/types.h

```diff
@@ -262,19 +262,19 @@ struct UnstructuredSetType : SetType {
   ~UnstructuredSetType();
 };
 
-struct LatticeLinkSetType : SetType {
-  /// Dimensionality of the lattice. This set must be of size d_1 x d_2 x ...
-  /// d_Nd x d. This type of set forces a LATTICE structure, such that the point
+struct GridSetType : SetType {
+  /// Dimensionality of the grid. This set must be of size d_1 x d_2 x ...
+  /// d_Nd x d. This type of set forces a GRID structure, such that the point
   /// in the underlying set at coordinate (... i_j, ...) neighbors points as
   /// coordinates (... i_j-1, ...) and (... i_j+1 ...), for all possible j. The
-  /// determination of boundary conditions is also delegated to the lattice link
+  /// determination of boundary conditions is also delegated to the grid edge
   /// set definition, though we assume periodic for now.
   size_t dimensions;
-  /// Underlying point set of the lattice. Elements of this edge set connect
-  /// neighboring grid points in the lattice.
-  IndexSet latticePointSet;
+  /// Underlying point set of the grid. Elements of this edge set connect
+  /// neighboring grid points in the grid.
+  IndexSet underlyingPointSet;
```

Comment:
I was hoping to make it absolutely clear that this set is defining an edge set _over_ a set of points, as opposed to the points themselves, so I would advocate for leaving this as-is. If it's too wordy, perhaps `underlyingPoints` would be better?